bt_automata/validator/reward_funcs.py

```diff
@@ -99,25 +113,40 @@ def get_rewards(
             bt.logging.debug("Simulation failed to produce a result.")
             return torch.FloatTensor([]).to(self.device)  # Or handle differently
 
+        # Pull the process times from the synapse responses
         process_times = [response.dendrite.process_time for uid, response in responses]
-        max_process_time, min_process_time = max(process_times), min(process_times)
-        rewards = np.zeros(256)
-        for uid, response in responses:
-            if response.array_data is None:
-                continue
-            result_accuracy = get_reward(gt_array, response)
-            # Linear normalization process time against the list of all miner process times
-            normalized_process_time = (
-                (response.dendrite.process_time - min_process_time)
-                / (max_process_time - min_process_time)
-                if max_process_time > min_process_time
-                else 0.0
-            )
-            # Include normalized process time in reward calculation
-            rewards[uid] = result_accuracy * 0.7 + 0.3 / (normalized_process_time + 1)
+
+        # Convert process times and result accuracies to tensors, if they aren't already
+        if not isinstance(process_times, torch.Tensor):
+            process_times = torch.tensor(process_times, dtype=torch.float32)
+        if not isinstance(result_accuracies, torch.Tensor):
+            result_accuracies = torch.tensor(result_accuracies, dtype=torch.float32)
+
+        # Normalize process times inversely so that lower times are better
+        normalized_process_times = (process_times - torch.min(process_times)) / (torch.max(process_times) - torch.min(process_times))
+        inverted_process_times = 1.0 - normalized_process_times  # Invert so higher times have lower scores
+        #breakpoint()
+
+        # Apply the sigmoid function to the inverted normalized process times
+        sigmoid_process_times = sigmoid(inverted_process_times, temperature, shift)
+
+        # Calculate accuracies for each response
+        accuracies = [get_accuracy(gt_array, response) for uid, response in responses]
+        accuracies_tensor = torch.tensor(accuracies, dtype=torch.float32)
+
+        # Weight the accuracy and speed, multiplying by result_accuracy to handle 0 accuracy case mathematically
+        rewards = accuracies * sigmoid_process_times
```

Comment:
This will throw an error. Use accurancies_tensor instead.

Reply:
Done